spinnman/spalloc/spalloc_transceiver.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2025 The University of Manchester
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import io
+import os
+import struct
+from typing import List, Union, BinaryIO, Optional, Tuple, cast, Dict
+
+import numpy
+from numpy import uint8, uint32
+
+from spinn_utilities.overrides import overrides
+
+from spinnman.transceiver.base_transceiver import BaseTransceiver
+from spinnman.connections.abstract_classes.connection import Connection
+from spinnman.processes.write_memory_process import _UNSIGNED_WORD
+from spinnman.model.diagnostic_filter import DiagnosticFilter
+
+from .spalloc_job import SpallocJob
+
+_ONE_WORD = struct.Struct("<I")
+
+
+class SpallocTransceiver(BaseTransceiver):
```

Comment:
I think this should Extend the Version5Transceiver.
This will make it more future proof to and Spin2 spalloc